tests/agentzh.t

```diff
@@ -113,7 +113,26 @@ print(cjson.encode(data))
 
 
 
-=== TEST 9: multiple calls to lua_cjson_new (1/2)
+=== TEST 9: cjson.array_mt is an alias to cjson.empty_array_mt
+--- lua
+local cjson = require "cjson"
+print(cjson.array_mt == cjson.empty_array_mt)
+--- out
+true
+
+
+
+=== TEST 10: array_mt behaves as empty_array_mt
+--- lua
+local cjson = require "cjson"
+local empty_arr = setmetatable({}, cjson.array_mt)
+print(cjson.encode({obj = {}, arr = empty_arr}))
+--- out
+{"arr":[],"obj":{}}
+
```

Comment:
Need to test non-empty tables with `cjson.array_mt` as well, including those use the hash table to store the array elements, as in

```
local new_tab = require "table.new"
local t = new_tab(0, 2)  -- only allocate the hash part
t[1] = "hello"
setmetatable(t, cjson.array_mt)
cjson.encode(t)  -- should be ["hello"] here
```